tests/test_parallel.py

```diff
@@ -0,0 +1,239 @@
+                6.772045449820714,
+                9.344911262612964,
+                9.772939589989000,
+                8.901837608843241,
+            ],
+        ),
+        (
+            True,  # Minimize: QCP
+            [
+                1.6578281551053196,
+                1.7175638492348284,
+                1.5946081072641485,
+                2.1234944939308220,
+                2.4894805175766606,
+            ],
+        ),
+    ],
+    ids=["no_minimize", "minimize"],
+)
+def test_prmsdwrapper_nosymm_protein(trps, minimize: bool, referenceRMSDs: List[float]):
+    mol0 = copy.deepcopy(trps[0])
+    mols = [copy.deepcopy(mol) for mol in trps[1:]]
+
+    RMSDs = prmsdwrapper(mol0, mols, symmetry=False, minimize=minimize, strip=False)
+
+    for RMSD, referenceRMSD in zip(RMSDs, referenceRMSDs):
+        np.testing.assert_allclose(RMSD, referenceRMSD, atol=1e-5)
+
+
+@pytest.mark.parametrize(
+    # Reference results obtained with OpenBabel
+    "minimize, referenceRMSDs",
+    [
+        (
+            True,  # Minimize: QCP + Isomorphism
+            [
+                0.476858,
+                1.68089,
+                1.50267,
+                1.90623,
+                1.01324,
+                1.31716,
+                1.11312,
+                1.06044,
+                0.965387,
+                1.37842,
+            ],
+        ),
+        (
+            False,  # No minimize: Isomorphism only
+            [
+                0.592256,
+                2.11545,
+                2.29824,
+                9.45773,
+                1.35005,
+                9.44356,
+                9.59758,
+                9.55076,
+                2.44067,
+                9.6171,
+            ],
+        ),
+    ],
+    ids=["minimize", "no_minimize"],
+)
+def test_prmsdwrapper_isomorphic(
+    docking_1cbr, minimize: bool, referenceRMSDs: List[float]
+) -> None:
+    molref = copy.deepcopy(docking_1cbr[0])
+    mols = [copy.deepcopy(mol) for mol in docking_1cbr[1:]]
+
+    RMSDs = prmsdwrapper(molref, mols, minimize=minimize, strip=True)
+
+    for RMSD, referenceRMSD in zip(RMSDs, referenceRMSDs):
+        np.testing.assert_allclose(RMSD, referenceRMSD, atol=1e-5)
+
+
+@pytest.mark.parametrize(
+    # Reference results obtained with OpenBabel
+    "minimize, referenceRMSD",
+    [(True, 0.476858), (False, 0.592256)],
+    ids=["minimize", "no_minimize"],
+)
+def test_prmsdwrapper_single_molecule(
+    docking_1cbr, minimize: bool, referenceRMSD: float
+) -> None:
+    molref = copy.deepcopy(docking_1cbr[0])
+    mols = copy.deepcopy(docking_1cbr[1])
+
+    RMSD = prmsdwrapper(molref, mols, minimize=minimize, strip=True)
+
+    np.testing.assert_allclose(RMSD[0], referenceRMSD, atol=1e-5)
+
+
+def test_prmsdwrapper_single_molecule_timeout(muparfostat) -> None:
+    mol1 = copy.deepcopy(muparfostat)
+    mol2 = copy.deepcopy(muparfostat)
+
+    with pytest.warns(
+        UserWarning, match=r"\d+ compounds failed to process successfully"
```

Comment:
This matches something that is always printed when there is an error (either timeout or a general `Exception`.

I tried to narrow it down to `1 chunks timed out` which is what I would expect for a timeout. However, it seems that the other branch is hit (and that is why there is no coverage on the `TimeoutError` branch:
```
>       with pytest.warns(
            UserWarning, match=r"1 chunks timed out"
        ):
E       Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) matching the regex were emitted.
E        Regex: 1 chunks timed out
E        Emitted warnings: [ UserWarning("1 compounds failed to process successfully and have been added as 'np.nan'. 1 compounds raised an error, 0 chunks timed out")].
```

Reply:
If I re-rise the exception captured by the `Exception` block, I get
```
>       raise result
E       AttributeError: 'Mol' object has no attribute 'strip'
```

Reply:
The issue is that this particular test was not properly updated after #118, where
the helper class
https://github.com/RMeli/spyrmsd/blob/8ce0edea593d86690132b47e6923a91de6d63bd8/tests/conftest.py#L18
was introduced.

The following change appears to fix it (with the correct match). Would you mind fixing it?

```diff
diff --git a/tests/test_parallel.py b/tests/test_parallel.py
index bbdd4c1..631d629 100644
--- a/tests/test_parallel.py
+++ b/tests/test_parallel.py
@@ -122,11 +122,11 @@ def test_prmsdwrapper_single_molecule(


 def test_prmsdwrapper_single_molecule_timeout(muparfostat) -> None:
-    mol1 = copy.deepcopy(muparfostat)
-    mol2 = copy.deepcopy(muparfostat)
+    mol1 = muparfostat.mol
+    mol2 = muparfostat.mol

     with pytest.warns(
-        UserWarning, match=r"\d+ compounds failed to process successfully"
+        UserWarning, match=r"1 chunks timed out"
     ):
         RMSD = prmsdwrapper(mol1, mol2, timeout=1e-3, num_workers=1)

```

Apologies for not spotting this issue early. This appears to be a classic example of the dangers of capturing the base class `Exception`. Do you think we can narrow that capture to avoid such issues? Or maybe it is better to not capture non-timeout exceptions at all?

Reply:
Thanks @Jnelen. Now it's the exception branch that [is not tested](https://app.codecov.io/gh/RMeli/spyrmsd/commit/73ed019a652c20b5adf21ffe7110988ef14824ef/blob/spyrmsd/parallel.py). Do you have an opinion on what I raided above?

Otherwise I'd just add a test where an exception is triggered. Given what happened here, I think using one of the molecules as `None` would do the trick. Then it's good to go in my end.